engine/access/rest/request_decorator.go

```diff
@@ -0,0 +1,145 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// a convenience wrapper around the http request to make it easy to read request query params
+type request struct {
+	*http.Request
+	expandFields map[string]bool
+	selectFields []string
+}
+
+// decorateRequest takes http request and applies functions to produce our custom
+// request object decorated with values we need
+func decorateRequest(r *http.Request) *request {
+	decoratedReq := &request{
+		Request: r,
+	}
+
+	if expandFields, found := middleware.GetFieldsToExpand(r); found {
+		decoratedReq.expandFields = sliceToMap(expandFields)
+	}
+
+	if selectFields, found := middleware.GetFieldsToSelect(r); found {
+		decoratedReq.selectFields = selectFields
+	}
+
+	return decoratedReq
+}
+
+func sliceToMap(values []string) map[string]bool {
+	valueMap := make(map[string]bool, len(values))
+	for _, v := range values {
+		valueMap[v] = true
+	}
+	return valueMap
+}
+
+func (rd *request) expands(field string) bool {
+	return rd.expandFields[field]
+}
+
+func (rd *request) selects() []string {
+	return rd.selectFields
+}
+
+func (rd *request) getVar(name string) string {
+	vars := mux.Vars(rd.Request)
+	return vars[name] // todo(sideninja) consider returning err if non-existing
+}
+
+func (rd *request) getQueryParam(name string) string {
+	return rd.Request.URL.Query().Get(name) // todo(sideninja) consider returning err if non-existing
```

Comment:
what would currently be returned if non-existing?

Reply:
Currently it will be an empty value.